Question_2.py

```diff
@@ -0,0 +1,112 @@
+# 2. You are given a list of person names. Your task is to find out the three most frequent and three least frequent names.
+
+# Input:
+# names = ['Smith', 'Johnson', 'Williams', 'Jones', 'Brown', 'Davis', 'Miller', 'Wilson', 'Moore', 'Taylor', 'Anderson', 'Thomas', 'Jackson', 'White']
+
+# Output:
+# o Names: ['Smith', 'Johnson', 'Williams', 'Jones', 'Brown', 'Davis', 'Miller',
+#             'Wilson', 'Moore', 'Taylor', 'Anderson', 'Thomas', 'Jackson', 'White'] 
+# o Name lengths: [5, 7, 8, 5, 5, 5, 6, 6, 5, 6, 8, 6, 7, 5]
+
+# o The three most frequent name lenghts are:
+# 6 names of length 5: ['Smith', 'Jones', 'Brown', 'Davis', 'Moore'] 
+# 4 names of length 6: ['Miller', 'Wilson', 'Taylor', 'Thomas']
+# 2 names of length 8: ['Williams', 'Anderson']
+
+# o The three least frequent name lenghts are:
+# 2 names of length 7: ['Johnson', 'Jackson']
+# 2 names of length 8: ['Williams', 'Anderson']
+# 4 names of length 6: ['Miller', 'Wilson', 'Taylor', 'Thomas']
+
+names= ['Smith', 'Johnson', 'Williams', 'Jones', 'Brown', 'Davis', 'Miller',
+        'Wilson', 'Moore', 'Taylor', 'Anderson', 'Thomas', 'Jackson', 'White'] 
+
+print('\n',names)
+
+
+# o Name lengths: [5, 7, 8, 5, 5, 5, 6, 6, 5, 6, 8, 6, 7, 5]
+name_lengths = []
+for i in names:
+    name_lengths.append(len(i))
+print("\nName lengths:", name_lengths,'\n')
+
+
+
+# ---------------------------------------------------------------------------------------------------------
+
+# o The three most frequent name lenghts are:
+# 6 names of length 5: ['Smith', 'Jones', 'Brown', 'Davis', 'Moore'] 
+# 4 names of length 6: ['Miller', 'Wilson', 'Taylor', 'Thomas']
+# 2 names of length 8: ['Williams', 'Anderson']
+
+dict = {}
+
+# empty dictionary to store the frequency count for each element (key), 
+# and its corresponding value is a list with all elements that have this key's value
+for name in names:
+    if len(name) in dict:
+        dict[len(name)] += 1
+    else:
+        dict[len(name)] = 1
```

Comment:
You can do here like:

dict[len(name)] = dict.get(len(name), 0) + 1

Also, please rename the `dict` to something else, as `dict` is reserved keyword in python which is used for `dict` class. 

Make sure you don't define variables that are reserved keyword in python.